File: JTime-android/src/main/java/com/ismail_s/jtime/android/MasjidPojo.java
package com.ismail_s.jtime.android;

import java.util.GregorianCalendar;

public class MasjidPojo {
    private String name;
    private GregorianCalendar fajrTime;
    private GregorianCalendar zoharTime;
    private GregorianCalendar asrTime;
    private GregorianCalendar magribTime;
    private GregorianCalendar eshaTime;

    public MasjidPojo(String name, GregorianCalendar fajrTime, GregorianCalendar zoharTime, GregorianCalendar asrTime, GregorianCalendar magribTime, GregorianCalendar eshaTime) {
        this.name = name;
        this.fajrTime = fajrTime;
        this.zoharTime = zoharTime;
        this.asrTime = asrTime;
        this.magribTime = magribTime;
        this.eshaTime = eshaTime;
    }

    public MasjidPojo(String name) {
        this.name = name;
    }

    public GregorianCalendar getFajrTime() {
        return fajrTime;
    }

    public void setFajrTime(GregorianCalendar fajrTime) {
        this.fajrTime = fajrTime;
    }

    public GregorianCalendar getZoharTime() {
        return zoharTime;
    }

    public void setZoharTime(GregorianCalendar zoharTime) {
        this.zoharTime = zoharTime;
    }

    public GregorianCalendar getAsrTime() {
        return asrTime;
    }

    public void setAsrTime(GregorianCalendar asrTime) {
        this.asrTime = asrTime;
    }

    public GregorianCalendar getMagribTime() {
        return magribTime;
    }

    public void setMagribTime(GregorianCalendar magribTime) {
        this.magribTime = magribTime;
    }

    public GregorianCalendar getEshaTime() {
        return eshaTime;
    }

    public void setEshaTime(GregorianCalendar eshaTime) {
        this.eshaTime = eshaTime;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public GregorianCalendar[] getTimes() {
        return new GregorianCalendar[]{fajrTime, zoharTime, asrTime, magribTime, eshaTime};
    }
}

File: JTime-android/src/main/java/com/ismail_s/jtime/android/MasjidPojo.kt
package com.ismail_s.jtime.android

import java.util.GregorianCalendar

class MasjidPojo {
    var name: String? = null
    var id: String = ""
    var address: String = ""
    var latitude: Double = 0.0
    var longitude: Double = 0.0
    var fajrTime: GregorianCalendar? = null
    var zoharTime: GregorianCalendar? = null
    var asrTime: GregorianCalendar? = null
    var magribTime: GregorianCalendar? = null
    var eshaTime: GregorianCalendar? = null

    constructor(name: String, fajrTime: GregorianCalendar, zoharTime: GregorianCalendar, asrTime: GregorianCalendar, magribTime: GregorianCalendar, eshaTime: GregorianCalendar) {
        this.name = name
        this.fajrTime = fajrTime
        this.zoharTime = zoharTime
        this.asrTime = asrTime
        this.magribTime = magribTime
        this.eshaTime = eshaTime
    }

    constructor(name: String) {
        this.name = name
    }

    constructor(name: String, id: String, address: String, latitude: Double, longitude: Double) {
        this.name = name
        this.id = id
        this.address = address
        this.latitude = latitude
        this.longitude = longitude
    }

    constructor()

    val times: Array<GregorianCalendar>
        get() = arrayOf(fajrTime!!, zoharTime!!, asrTime!!, magribTime!!, eshaTime!!)
}

enum class SalaahType {
    FAJR, ZOHAR, ASR, MAGRIB, ESHA
}
